operator/src/network/controller.rs

```diff
@@ -473,27 +476,52 @@ async fn apply_bootstrap_job(
 }
 
 async fn update_peer_info(
-    cx: Arc<Context<impl RpcClient>>,
+    cx: Arc<Context<impl IpfsRpcClient>>,
     ns: &str,
     network: Arc<Network>,
     status: &mut NetworkStatus,
 ) -> Result<(), Error> {
-    println!("update_peer_info {:?}", status);
+    // Check status of all ceramic peers first
     for index in 0..status.replicas {
-        if status.peers.iter().any(|info| info.index == index) {
+        if status.peers.iter().any(|info| info.index() == index) {
             // Skip peers we already know
             continue;
         }
-        let info = match cx.rpc_client.peer_info(ns, index).await {
+        let ipfs_rpc_addr = ceramic_peer_ipfs_rpc_addr(ns, index);
+        let info = match cx.rpc_client.peer_info(index, ipfs_rpc_addr).await {
             Ok(res) => res,
             Err(err) => {
-                trace!(%err, index, "failed to get peer id and mulitaddrs for peer");
+                trace!(%err, index, "failed to get peer info for ceramic peer");
                 continue;
             }
         };
-        status.peers.push(info);
+        let ceramic_addr = ceramic_addr(ns, index);
+        status.peers.push(Peer::Ceramic(CeramicPeerInfo {
+            ceramic_addr,
+            index: info.index,
+            peer_id: info.peer_id,
+            ipfs_rpc_addr: info.ipfs_rpc_addr,
+            p2p_addrs: info.p2p_addrs,
+        }));
     }
+    // ready_replicas count should not include other non ceramic peers
     status.ready_replicas = status.peers.len() as i32;
+
+    // Add extra peers, using negative peer indexes
+    {
+        // CAS IPFS peer
+        let ipfs_rpc_addr = format!("http://{CAS_IFPS_SERVICE_NAME}-0.{CAS_IFPS_SERVICE_NAME}.{ns}.svc.cluster.local:{CAS_SERVICE_IPFS_PORT}");
```

Comment:
it was mentioned before to have the option of passing a remote CAS, looks like that would still be easy to configure here